core/src/navigation/ContextSwitcher.html

```diff
@@ -0,0 +1,194 @@
+        {#if !selectedLabel}{config.emptyLabel}{/if}
+        {#if selectedLabel}{selectedLabel}{/if}
+      </button>
+    </div>
+    <div class="fd-popover__body" aria-hidden="{dropDownStatesNegated.CONTEXT_SWITCHER_POPOVER || true}" id="CONTEXT_SWITCHER_POPOVER">
+      <nav class="fd-menu">
+        {#if actions && actions.length}
+        <ul class="fd-menu__list fd-menu__list--top">
+          {#each actions as node}
+          {#if node.position === 'top' || !['top', 'bottom'].includes(node.position)}
+          <li><a class="fd-menu__item" on:click="onActionClick(node)">{node.label}</a></li>
+          {/if}
+          {/each}
+        </ul>
+        {/if}
+
+        <ul class="fd-menu__list" id="context_menu_middle">
+        {#if options.length === 0}
+          <li><div class="fd-spinner"><div></div></div></li>
+        {/if}
+        {#if options.length}
+          {#each options as node}
+          <li><a class="fd-menu__item" on:click="goTo(node.path)">
+          {#if node.label}{node.label}{/if}
+          {#if !node.label}
+            {#await getMatchingNodeName(options, config.fallbackLabelResolver, node.id)}
+            {:then label}
+              {label}
+            {/await}
+          {/if}
+          </a></li>
+          {/each}
+        {/if}
+        </ul>
+        {#if actions && actions.length}
+        <ul class="fd-menu__list fd-menu__list--bottom">
+          {#each actions as node}
+          {#if node.position === 'bottom'}
+          <li><a class="fd-menu__item" on:click="onActionClick(node)">{node.label}</a></li>
+          {/if}
+          {/each}
+        </ul>
+        {/if}
+      </nav>
+    </div>
+  </div>
+</div>
+{/if}
+
+<script type="text/javascript">
+  import { ContextSwitcherHelpers } from './services/context-switcher.js';
+  import { getConfigValue, getConfigValueAsync, getConfigValueFromObjectAsync, getConfigBooleanValue } from '../services/config.js';
+  import { getNegatedBoolString, trimLeadingSlash } from '../utilities/helpers.js';
+  import * as Routing from '../services/routing.js';
+
+  export default {
+    async oncreate() {
+      const contextSwitcherConfig = getConfigValue('navigation.contextSwitcher');
+      this.set({
+        contextSwitcherEnabled: !!contextSwitcherConfig,
+        config: contextSwitcherConfig
+      });
+
+      if (contextSwitcherConfig) {
+        const actions = await getConfigValueAsync('navigation.contextSwitcher.actions');
+        this.set({
+          actions,
+          fallbackLabelResolver: contextSwitcherConfig.fallbackLabelResolver
+        });
+        // options are loaded lazy by default
+        if (contextSwitcherConfig.lazyloadOptions === false) {
+          await this.fetchOptions();
+        }
+      }
+
+      await this.setSelectedContext(Routing.getCurrentPath());
+      Routing.addRouteChangeListener(() => this.setSelectedContext(Routing.getCurrentPath()));
+    },
+    data() {
+      return {
+        dropDownStates: {},
```

Comment:
It would be nice to have an initialisation of all keys here, I see above we are setting some other keys on component creation.